python/az/aro/azext_aro/custom.py

```diff
@@ -134,9 +136,45 @@ def aro_create(cmd,  # pylint: disable=too-many-locals
                        parameters=oc)
 
 
-def aro_delete(client, resource_group_name, resource_name, no_wait=False):
+def aro_delete(cmd, client, resource_group_name, resource_name, no_wait=False):
     # TODO: clean up rbac
 
+    oc = None
+    try:
+        oc = client.get(resource_group_name, resource_name)
+    except CloudError:
+        pass
+
+    if oc:
+        master_subnet = oc.master_profile.subnet_id
+        worker_subnets = [w.subnet_id for w in oc.worker_profiles]
+
+        master_parts = parse_resource_id(master_subnet)
+        vnet = resource_id(
+            subscription=master_parts['subscription'],
+            resource_group=master_parts['resource_group'],
+            namespace='Microsoft.Network',
+            type='virtualNetworks',
+            name=master_parts['name'],
+        )
+
+        aad = AADManager(cmd.cli_ctx)
+
+        client_id = oc.service_principal_profile.client_id
+
+        client_sp = aad.get_service_principal(client_id)
+        if not client_sp:
+            client_sp = aad.create_service_principal(client_id)
+
+        rp_client_id = FP_CLIENT_ID
+        rp_client_sp = aad.get_service_principal(rp_client_id)
+
+        # Customers frequently remove these permissions, then cannot delete their
+        # clusters. Hence, verify this before attempting deletion.
+        for sp_id in [client_sp.object_id, rp_client_sp.object_id]:
+            assign_contributor_to_vnet(cmd.cli_ctx, vnet, sp_id)
+            assign_contributor_to_routetable(cmd.cli_ctx, worker_subnets + [master_subnet], sp_id)
```

Comment:
@ehashman I've come around to your error handling here.  I think failures on all of these steps should be non-fatal: a user might have perms to delete the cluster but not mess with aad/rbac - the delete might be perfectly feasible and we shouldn't prevent them from deleting